core/src/main/java/org/infinispan/stream/impl/tx/TxClusterStreamManager.java

```diff
@@ -106,7 +106,7 @@ public void forgetOperation(Object id) {
          Supplier<Map.Entry<Address, IntSet>> segments, Set<K> keysToInclude, IntFunction<Set<K>> keysToExclude,
          boolean includeLoader, boolean entryStream, Iterable<IntermediateOperation> intermediateOperations) {
 
-      if (ctx.getLookedUpEntries().isEmpty()) {
+      if (ctx.lookedUpEntriesCount() != 0) {
```

Comment:
Just happen to be peeking around and found this is a bug, this should really be `== 0` instead.

Reply:
Thanks for fixing it in https://github.com/infinispan/infinispan/pull/7016 @wburns !